src/main/java/software/amazon/cloudformation/proxy/package-info.java

```diff
@@ -0,0 +1,461 @@
+ * resources are not available, e.g. when associating KMS Key with CloudWatch
+ * Log group, the key might not have propogated to region leading to failures to
+ * associate with the log group. This framework provides developers the
+ * facilities to deal with all of these situations using a simple functional
+ * model.
+ * <p>
+ * The framework handles the following for developers:
+ * <ol>
+ * <li>Taking care of communicating all progress events consistently with
+ * CloudFormation including appropriate handshakes</li>
+ * <li>Handles all client side error reporting correctly with appropriate
+ * failure codes and errors messages</li>
+ * <li>Handles common AWS service exceptions like Throttle and several other
+ * retry errors based on HTTP codes</li>
+ * <li>Can auto-renew STS token expiry transparently if needed or schedule
+ * callback from CloudFormation requesting new credentials</li>
+ * <li>Automatically handles AWS Lambda runtime constraints and schedules
+ * callbacks when we near max computation time. Auto adjusts behavior based on
+ * the runtime the handler is bound other modes for operating</li>
+ * <li>Memoize all service call requests, responses, stabilization and retry
+ * handling that ensures continuations of handler compute start from where we
+ * left off for callbacks.</li>
+ * <li>Provides default flat retry model for calls of 5 seconds with max timeout
+ * of 2 minutes for all API calls. Allows for sensible defaults to be specified
+ * for retry strategy by integrator on a per API call basis. Allows framework to
+ * dynamically override backoff behavior for helping with emergencies with
+ * downstream service latency and other situations defined by customer need</li>
+ * <li>Provides developers with a simple sequential pattern of progress chaining
+ * while doing all the heavy lifting above. Provides most common sensible
+ * defaults with override capabilities for integrator to influence as needed.
+ * Covers power users and simple users alike.</li>
+ * <li>Brings consistent model across all service integrations across AWS
+ * services on GitHub hosted code base.</li>
+ * </ol>
+ * <p>
+ * <b><u>Anatomy of an AWS Web Service call</u></b>
+ * <p>
+ * Most AWS web service API calls follows a typical pattern shown below
+ * <ol>
+ * <li>Initiate the call context in preparation of the API</li>
+ * <li>Transform the incoming CFN resource model properties to the underlying
+ * service API request</li>
+ * <li>Make the service call</li>
+ * <li>Handle stabilization (optional if we need the resource to be in specific
+ * state before we apply next part of the state)</li>
+ * <li>Handle errors (optional, framework already handles most errors and
+ * retries ones that can retried and communicates error codes when
+ * appropriate)</li>
+ * <li>Proceed with progressing to the next sequence or indicate successful
+ * completion.</li>
+ * </ol>
+ *
+ * When integrating with AWS APIs, here are a few common patterns across Create,
```

Comment:
Maybe reframe this around how this pattern helps conform to "The Contract ™️ " (https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html)